### DISCUSSION TITLE
Enable push notifications for all in-app notifications

### DISCUSSION THREAD

**Discussion by JacksonCampbell99** [community/community#7229] [Mobile]
I can't tell you how confused I get on what is supposed to notify me and what is not. I have every push notification setting on in the app and in the system notification settings for the app. If it's your issue or discussion, you will get notified in the app; if they happened to @ you then you will get a push notification. That encourages unnecessarily @ ing people, such as when I am the OP and someone is just commenting on the post (OP gets in-app notifications for comments). Knowing how notifications work, some people will comment and @ me, and some will just reply without @ ing me, which should be totally reasonable. @ ing should only be considered necessary when on a thread with multiple people interacting, not even a thread with only OP and another person back and forth.

**Comment by brianlovin** [comment]
Thanks for the note @JacksonCampbell99 — I can see how this would be annoying in threads with just two people. Right now we don't have plans to support push notifications for _all_ comments on issues, but I'll pass this along to our team to think about how we might approach this without potentially overwhelming people with push notifications if they have a lot of simultaneous conversations happening.

**Reply by JacksonCampbell99** [reply]
Thank you. And when people are replying like you just did on my post. I only received a notification from your comment because you tagged me, which should be unnecessary in this scenario.

**Reply by leoskyrocker** [reply]
It'd be REALLY great if the mobile app can support that. I've been relying a lot on the push notifications from the app rather than the emails, but I've been missing some notifications on when people replied to my code review comments, or when they finished reviewing/commenting on my PR. It'd be really great if I can fully rely on the mobile app to work efficiently with PRs! It narrows the gap between the back and forth loop so much.

**Reply by ZuOliver** [reply]
![template-editor2-dffd8f7c94ea3ed730405de939aac5ead86464d4b591bbda1884be4f7b4571e1.png](https://user-images.githubusercontent.com/93783622/142983980-214f531a-6706-4224-bee6-c7571ea0081b.png)

**Reply by JacksonCampbell99** [reply]
@@brianlovin If there were proper notifications like I'm advocating for, then I would be able to keep up with the comments here without being notified through email to have me go open the app and check notifications.

**Reply by GitSparTV** [reply]
1.5 year have been passed since I requested this when email was a way to request things.

Is it really cheaper to send me an email message for every event in GitHub than sending a notification? If I watch every repository I read all issues even if I'm not mentioned. After I read the thread I have to clean clog of garbage in my inbox from GitHub

**Reply by ZuOliver** [reply]
Vocês são nojentos você quer trabalhar no mc Donald? Se quiser agente troca d lugar e continuamos amigos Eu também não aguento mais apagar promoções de lojas internacionais no meu email é whtssap

**Reply by ZuOliver** [reply]
Porque nao Fazem comércio no país de vocês?

**Reply by aak-lear** [reply]
> Thanks for the note @JacksonCampbell99 — I can see how this would be annoying in threads with just two people. Right now we don't have plans to support push notifications for _all_ comments on issues, but I'll pass this along to our team to think about how we might approach this without potentially overwhelming people with push notifications if they have a lot of simultaneous conversations happening.

@brianlovin 

We really need a way to instantly receive notifications about PRs updates, new comments and commits.

Now the only way to receive such notifications is sending by email.

This is due to two inconveniences:

1) It's very tedious to often clean email from dozens of notifications from GitHub

2) Opening an email through a quick view does not always allow you to understand what changes have occurred. And you have to open the email client and read the  full letter

Native pushes on iOS and Android can solve both problems very well:

1) When you receive push, you can immediately clear it

2) If you can't understand, what's happened, you can tap to notification, and it will open the GitHub Mobile, where you can see all changes with full app functionality


If you are worried about potentially overwhelming people, then you should know, that receiving notifications on email is the same thing, but people also have to clean their emails frequently

**Reply by DavidJH2** [reply]
Could you allow the user to decide if they are "overwhelmed" or not

I would like to have the option to get ALL push notifications on a repository

**Reply by ghost** [reply]
I was searching for ages through the settings trying to work out why I wasn't getting push notifications for new notifications, and now I know why.

Please support push notifications on ALL notifications, not just mentions.

It's made app push notifications near useless to me, having to fall back to email notifications due to being unreliable by design.

**Reply by GitSparTV** [reply]
Celebrating 1.5 years since my message of celebrating 1.5 years of this feature not being implemented
![image](https://github.com/community/community/assets/5685050/d3c1cf38-3810-4a56-88dd-1ec3c2b9fbe5)

**Reply by Hacksore** [reply]
While this is an issue I heavily rely on filters to keep my inbox clean.

https://gist.github.com/ldez/bd6e6401ad0855e6c0de6da19a8c50b5

**Reply by DavidJH2** [reply]
Anyone know how to hack GitHub so that you can get Push notifications?

**Comment by ZuOliver** [comment]
Não estou entendendo Parei aqui decorrente Aplicativo Uber .Tô em busca de adicionar os serviços uber Flash e uberflash moto As minhas lojas e aplicativos on line E Também a Tiny ERP. ALGUÉM ME INDICA UM LINK EM PORTUGUÊS BRASIL ?@uber

**Comment by Kaminem202** [comment]
> Thanks for the note @JacksonCampbell99 — I can see how this would be annoying in threads with just two people. Right now we don't have plans to support push notifications for _all_ comments on issues, but I'll pass this along to our team to think about how we might approach this without potentially overwhelming people with push notifications if they have a lot of simultaneous conversations happening.

**Reply by JacksonCampbell99** [reply]
It's called you have a setting and people can turn it on or off just like the other ones. It's not overwhelming people to add an option. It's overwhelming to find out you missed a bunch of conversations because the app wouldn't notify you.

**Comment by DavidJH2** [comment]
> It's called you have a setting and people can turn it on or off just like the other ones. It's not overwhelming people to add an option. It's overwhelming to find out you missed a bunch of conversations because the app wouldn't notify you.

Whish I could upvote this a million times...
(more, really)

**Reply by sammyke007** [reply]
+1!

**Comment by Hacksore** [comment]
Dropping this thread here for visibility 🙏

https://github.com/orgs/community/discussions/106727#discussioncomment-10431242

**Comment by jo-chemla** [comment]
Also in favor of this behavior. 

Notifications on github mobile are already very configurable. So it would be great to add a toggle to allow users to get notified for all inbox notifications. This should include comments on issues, prs, discussions, anything the user has subscribed to, without necessary direct mention - simply replicating the email notification within mobile. 

Default behavior could be toggling this setting off to avoid overflooding users, overcoming raised concern in this thread.
